### PR TITLE
Fix typescript errors in recently added tests

### PR DESCRIPTION
**Pull request by Stefie**
I noticed that tslint was throwing some errors on recently added tests. This fixes them.

### CODE REVIEW

**Comment by Stefie** [comment]
Hmm, they used to be. I remember that when I did my big JS -> TS refactor PR  the pipeline failed on every tslint error in the e2e folders. So something must have changed recently...

File: packages/api-derive/test/e2e/promise.spec.ts [hunk 1]
```diff
@@ -34,7 +35,7 @@ describe.skip('derive e2e', () => {
   });
 
   it('subscribes to newHead, retrieving the actual validator', (done) => {
-    return api.derive.chain.subscribeNewHead(({ author }) => {
+    return api.derive.chain.subscribeNewHead(({ author }: HeaderExtended) => {
```
Comment (amaurymartiny) [inline packages/api-derive/test/e2e/promise.spec.ts]: Is tslint throwing on these too? This means the derive typings didn't work

Comment (Stefie) [reply packages/api-derive/test/e2e/promise.spec.ts]: The `api.derive.***` in general or this specific one? `api.derive.***` is a 100% yes, for `api.derive.chain.subscribeNewHead(({ author })` I'm only 90% sure, I'd have to roll back and double check.

**Comment by Stefie** [comment]
```
packages/api-derive/test/e2e/promise.spec.ts:37:49 - error TS7031: Binding element 'author' implicitly has an 'any' type.

37     return api.derive.chain.subscribeNewHead(({ author }) => {
                                                   ~~~~~~

packages/api-derive/test/e2e/promise.spec.ts:47:38 - error TS7006: Parameter 'fees' implicitly has an 'any' type.

47     return api.derive.balances.fees((fees) => {
                                        ~~~~

packages/api-derive/test/e2e/promise.spec.ts:55:89 - error TS7006: Parameter 'balance' implicitly has an 'any' type.

55     return api.derive.balances.all('5FLSigC9HGRKVhB9FiEo4Y3koPsNmBmLJbpXg2mp1hXcS59Y', (balance) => {
                                                                                           ~~~~~~~

packages/api-derive/test/e2e/promise.spec.ts:65:89 - error TS7006: Parameter 'balance' implicitly has an 'any' type.

65     return api.derive.balances.all('5F7BJL6Z4m8RLtK7nXEqqpEqhBbd535Z3CZeYF6ccvaQAY6N', (balance) => {
                                                                                           ~~~~~~~

packages/api-derive/test/e2e/promise.spec.ts:77:37 - error TS7006: Parameter 'info' implicitly has an 'any' type.

77     return api.derive.session.info((info) => {
                                       ~~~~

packages/api-derive/test/e2e/promise.spec.ts:90:48 - error TS7006: Parameter 'info' implicitly has an 'any' type.

90     return api.derive.staking.info(accountId, (info) => {
                                                  ~~~~

packages/api-derive/test/e2e/rx.spec.ts:115:48 - error TS7006: Parameter 'balances' implicitly has an 'any' type.

115         api.derive.balances.all(ID).subscribe((balances) => {
                                                   ~~~~~~~~

packages/api-derive/test/e2e/rx.spec.ts:133:47 - error TS7006: Parameter 'fees' implicitly has an 'any' type.

133         api.derive.balances.fees().subscribe((fees) => {
                                                  ~~~~

packages/api-derive/test/e2e/rx.spec.ts:150:50 - error TS7006: Parameter 'blockNumber' implicitly has an 'any' type.

150         api.derive.chain.bestNumber().subscribe((blockNumber) => {
                                                     ~~~~~~~~~~~

packages/api-derive/test/e2e/rx.spec.ts:160:59 - error TS7006: Parameter 'blockNumber' implicitly has an 'any' type.

160         api.derive.chain.bestNumberFinalized().subscribe((blockNumber) => {
                                                              ~~~~~~~~~~~

packages/api-derive/test/e2e/rx.spec.ts:170:53 - error TS7006: Parameter 'numberLag' implicitly has an 'any' type.

170         api.derive.chain.bestNumberLag().subscribe((numberLag) => {
                                                        ~~~~~~~~~

packages/api-derive/test/e2e/rx.spec.ts:181:55 - error TS7006: Parameter 'headerExtended' implicitly has an 'any' type.

181         api.derive.chain.getHeader('TODO').subscribe((headerExtended) => {
                                                          ~~~~~~~~~~~~~~

packages/api-derive/test/e2e/rx.spec.ts:191:56 - error TS7006: Parameter 'headerExtended' implicitly has an 'any' type.

191         api.derive.chain.subscribeNewHead().subscribe((headerExtended) => {
                                                           ~~~~~~~~~~~~~~

packages/api-derive/test/e2e/rx.spec.ts:202:57 - error TS7006: Parameter 'progress' implicitly has an 'any' type.

202         api.derive.session.sessionProgress().subscribe((progress) => {
                                                            ~~~~~~~~

packages/api/src/checkTypes.manual.ts:34:44 - error TS7006: Parameter 'header' implicitly has an 'any' type.

34   await api.derive.chain.subscribeNewHead((header) => {
``
`

**Pull request merged**